doc/fundamentals/append_overwrite.md

```diff
@@ -0,0 +1,616 @@
+
+This can have important consequences for downstream processing.  For example, let's say you have that same file named "count.txt" consisting of 5 lines that we used in the previous example
+```
+One
+Two
+Three
+Four
+Five
+```
+As discussed prior, loading that file into Pachyderm using `--split` will result in five files in a directory named "count.txt" in the input repo, each of which will have the following contents
+```
+count.txt/0000000000000000: One
+count.txt/0000000000000001: Two
+count.txt/0000000000000002: Three
+count.txt/0000000000000003: Four
+count.txt/0000000000000004: Five
+```
+This would result in five datums being processed in any pipelines that use this repo.
+
+Now, modify that file by inserting the word "Zero" on the first line.
+```
+Zero
+One
+Two
+Three
+Four
+Five
+```
+Let's upload it to Pachyderm using `--split` and `--overwrite`.
+```
+pachctl put-file line-data master count.txt -f ./count.txt --split line --overwrite
+```
+The input repo will now look like this
+```
+count.txt/0000000000000000: Zero
+count.txt/0000000000000001: One
+count.txt/0000000000000002: Two
+count.txt/0000000000000003: Three
+count.txt/0000000000000004: Four
+count.txt/0000000000000005: Five
+``` 
+As far as Pachyderm is concerned, every single file existing has changed, and a new file has been added.  Six datums would be processed by a downstream pipelines.
+
+It's important to remember that what looks like a simple upsert can be a kind of a [fencepost error](https://en.wikipedia.org/wiki/Off-by-one_error#Fencepost_error).  Being "off by one line" in your data can be expensive, consuming processing resources you didn't intend to spend.
+
+### Datums in Pachyderm pipelines
+
+The "datum" is the fundamental unit of data processing in Pachyderm pipelines.  It is defined at the file level and filtered by the "globs" you specify in your pipelines.  *What* makes a datum is defined by you.  How do you do that?
+
+When creating a pipeline, you can specify one or more input repos.  Each of these will contain files.  Those files are filtered by the "glob" you specify in the pipeline's defnition, along with the input operators you use.  That determines how the datums you want your pipeline to process appear in the pipeline: globs and input operators, along with other pipeline configuration operators, specify how you would like those datums orchestrated across your processing containers. Pachyderm Pipeline System (pps) processes each datum individually in containers in pods, using Pachyderm File System (pfs) to get the right data to the right code at the right time and merge the results. 
+
+To summarize:
```

Comment:
Documentation doesn't follow the "don't repeat yourself" dictum.  It's best to start a section that goes into advanced topics with a short review of what's been covered, and not distract the reader by asking them to "go here to learn this".  It is worthwhile to include a link to the detailed 101 doc if users want to follow it, but it's good to make each one self-contained rather than having folks keep clicking to get what they want.

Reply:
fair point. I'm ok with that as long as it doesn't go so far as you'll lose the reader with things they already know.

Reply:
You always fast forwarded through the "previously on The X Files", didn't you?